wolvic/browser/vr/wvr_manager.cc

```diff
@@ -536,26 +619,21 @@ bool WvrManager::WebVrCanAnimateFrame() {
 void WvrManager::GetFrameData(
     device::mojom::XRFrameDataRequestOptionsPtr options,
     device::mojom::XRFrameDataProvider::GetFrameDataCallback callback) {
-  if (!get_frame_data_callback_.is_null()) {
-    DLOG(WARNING) << ": previous get_frame_data_callback_ was not used yet";
-    frame_data_receiver_.ReportBadMessage(
-        "Requested VSync before waiting for response to previous request.");
-    ClosePresentationBindings();
-    return;
-  }
-
-  pending_time_ = base::TimeTicks();
-  get_frame_data_callback_ = std::move(callback);
-  WebXrTryStartAnimatingFrame();
-}
-
-void WvrManager::WebXrTryStartAnimatingFrame() {
   DCHECK(IsOnWvrThread());
+  CHECK(!graphics_->webxr_surface_size().IsEmpty());
 
   if (!WebVrCanAnimateFrame()) {
-    return;
+    // We bind this as a post task so that whatever processing is run when we
+    // attempt to get new frame data can complete before the pending
+    // GetFrameData call actually happens.
+    pending_getframedata_ = base::BindPostTask(
```

Comment:
It would be good to add a CHECK for !pending_get_framedata_. My understanding is that there can only be one pending call. But in case we find out we need more, then we would need to add a queue for processing getframedata_ calls?